src/xdp/program.c

```diff
@@ -425,6 +494,19 @@ XdpParseFrame(
         Cache->TransportPayload.BufferDataOffset = Offset;
         Cache->TransportPayload.IsFragmentedBuffer = FALSE;
         Cache->TransportPayloadValid = TRUE;
+    } else if (IpProto == IPPROTO_TCP) {
+        if (Buffer->DataLength < Offset + sizeof(*Cache->TcpHdr) ||
+            Buffer->DataLength < Offset + TCP_HDR_LEN_TO_BYTES(((TCP_HDR *)&Va[Offset])->th_len)) {
```

Comment:
We should try to avoid double fetching `TCP_HDR_LEN_TO_BYTES(((TCP_HDR *)&Va[Offset])->th_len)` since the packet buffer may be shared with user mode. The existing code isn't perfectly resilient to double fetches, but I'd like to try adding more.

Reply:
If we remove the options handling entirely, how do we advance to the TCP payload for CIBIR rules to parse?